### DISCUSSION TITLE
Is there a better way to show information?

### DISCUSSION THREAD

**Discussion by Antonio-Laguna** [johnlindquist/kit#146] [Q&A] [answered]
I've been toying with script kit to handle domotic devices through the Eedomus Gateway API. So far so good. Sometimes I just want to check my daughter's bedroom temperature and I was thinking that I would love a less intrusive way to show information that's not `show`.

I've tried a few things but it just goes away in a blink. And show can be intrusive and easy to leave open as soon as you click elsewhere on the screen.

![Apr-15-2021 10-14-30](https://user-images.githubusercontent.com/946645/114837712-4b7c5880-9dd4-11eb-9d26-68b447356abb.gif)

Also, args seem to leave the `value` as the `placeholder` somehow.

**Comment by johnlindquist** [comment, marked as answer]
Great question!

I'm not exactly sure what your code looks like, but the only way to "pause" a script is to use one of the helpers (otherwise the script runs to completion and forces the prompt closed). For example:

```js
let result = await arg()

await arg(result)
```

That second `arg` will just hang there waiting for either input, "blur", or escape. 

If you to display HTML:
```js
let result = await arg()

await arg("", `<h1>${result}</h1>`)
```

**Comment by Antonio-Laguna** [comment]
That works like a charm!

The code should anyone is interested is:


```js
const USER = await env('EEDOMUS_USER');
const SECRET = await env('EEDOMUS_SECRET');
const getUrl = suffix => `http://api.eedomus.com/get?api_user=${USER}&api_secret=${SECRET}${suffix}`;
const peripherals = await get(getUrl('&action=periph.list'), {
  responseEncoding: 'latin1'
});

const query = await arg('Type', ['Temperature', 'Lights']);

switch (query) {
  case 'Temperature': {
    const choices = peripherals.data.body
      .filter(p => p.usage_name.includes('Temp'))
      .map(p => ({ name: p.room_name, value: p.periph_id }));
    const peripheral = await arg('Sensor', choices);
    const value = await get(getUrl(`&action=periph.value&periph_id=${peripheral}`), {
      responseEncoding: 'latin1'
    });

    await arg('', `<h1>${value.data.body.last_value}</h1>`);
  }
}
```

**Reply by johnlindquist** [reply]
Very cool script! Thanks so much for sharing
